File: Selection-sort/java/SelectionSort.java
public class SelectionSort {
    public static void selectionSort(int[] arr, int n){
        for(int i = 0; i<n - 1;i++){
            int min = i;
            for(int j = i + 1; j< n;j++){
                if (arr[j] < arr[min]) {
                    min = j;
                }
            }
            int temp = arr[min];
            arr[min] = arr[i];
            arr[i] = temp;
        }
    }
    public static void main (String[] args){
        int[] arr = {64, 34, 25, 12, 22, 11, 9};
        System.out.println("Unsorted array:");
        for (int i = 0; i <arr.length; i++)
            System.out.print(arr[i] + " ");
        System.out.println();
        selectionSort(arr, arr.length);
        System.out.println("Sorted array:");
        for (int i = 0; i <arr.length; i++)
            System.out.print(arr[i] + " ");
        System.out.println();
    }
}

File: Selection-sort/go/SelectionSort.go
package main

import (
	"fmt"
)

var arr = []int{64, 34, 25, 12, 22, 11, 90} 

func SelectionSort(arr []int, n int) {
    for i := 0; i < n-1; i++ {
        min := i
        for j := i + 1; j < n; j++ {
            if arr[j] < arr[min] {
                min = j
            }
        }
        temp := arr[min]
        arr[min] = arr[i]
        arr[i] = temp
    }
}


func main() {
	fmt.Println("Unsorted array:")
	fmt.Println(arr)
	SelectionSort(arr, len(arr))
	fmt.Println("Sorted array:")
	fmt.Println(arr)
}